libr/parse/p/parse_x86_pseudo.c

```diff
@@ -358,12 +364,58 @@ static inline void mk_reg_str(const char *regname, int delta, bool sign, bool at
 	r_strbuf_free (sb);
 }
 
-static bool varsub (RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data, char *str, int len) {
-	RList *bpargs, *spargs;
+static const RList *__args(RParse *p, RAnalFunction *f, int type) {
+#if CACHE_VARS
+	static RAnalFunction *old_f = NULL;
+	static RList *__cache_spargs_list = NULL;
+	static RList *__cache_bpargs_list = NULL;
+	if (type == -1 && !p && !f) {
```

Comment:
maybe you meant `||` ? I'm not sure this condition can ever be met otherwise, as at https://github.com/radareorg/radare2/pull/15984/files#diff-18af017c7f2f2d244eb96f6f6daa48efR407 p is probably *not null*.